specification/app/resource-manager/Microsoft.App/preview/2023-11-02-preview/AppSpaceResources.json

```diff
@@ -0,0 +1,433 @@
+          "Patch Container App": {
+            "$ref": "./examples/AppSpaceResources_Patch.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DeploymentInformation": {
+      "description": "App Space Resource Deployment Information - used to track the deployment information for an App Space Resource's tracked ARM resource",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "ARM Id of the deployment",
+          "type": "string"
+        },
+        "provisioningState": {
+          "description": "Provisioning State for the deployment",
+          "type": "string"
+        }
+      }
+    },
+    "RepositoryInformation": {
+      "description": "App Space Resource Repository Information - used to track the github repository for an App Space Resource's tracked ARM resource",
+      "type": "object",
+      "properties": {
+        "url": {
+          "description": "url for the repository",
+          "type": "string"
+        },
+        "branch": {
+          "description": "branch in the repository",
+          "type": "string"
+        }
+      }
+    },
+    "Object": {
+      "type": "object",
+      "properties": {}
+    },
+    "AppSpaceResource": {
+      "description": "App Space Resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "extendedLocation": {
+          "$ref": "./CommonDefinitions.json#/definitions/ExtendedLocation"
+        },
+        "properties": {
+          "description": "AppSpaceResource resource specific properties",
+          "type": "object",
+          "properties": {
+            "resourceId": {
+              "description": "ARM resource Id of the tracked resource.",
+              "type": "string"
+            },
+            "friendlyName": {
+              "description": "Friendly name of the App Space Resource.",
+              "type": "string"
+            },
+            "resourceLocation": {
+              "description": "Location of the tracked ARM resource. Can be different from the location of the app space resource.",
+              "type": "string"
+            },
+            "repo": {
+              "$ref": "#/definitions/RepositoryInformation",
+              "description": "Optional App Space Resource Repository Information"
+            },
+            "deployment": {
+              "$ref": "#/definitions/DeploymentInformation",
+              "description": "Optional App Space Resource Deployment Information"
+            },
+            "isFavorite": {
+              "description": "Boolean marking this App Space Resource as a favorite",
+              "type": "boolean"
+            },
+            "metadata": {
+              "$ref": "#/definitions/Object",
+              "description": "Property bag to add properties used by Portal."
```

Comment:
this is not allowed , you need to define all the json schema exactly

Reply:
I've seen Microsoft.Web do this in their swagger - our portal team has had things that are just a property bag in the past and asked for that in the api spec. Can you possible give me an example of how to do this without having an object definition (which is what I saw in Microsoft.Web's swagger)? Thank you!

Reply:
Don't use Microsoft.Web as an example. The oldest RPs generally have the least ARM-compliant APIs. You can use an array of objects or strings, otherwise use an object. Unschematized objects are only allowed for contracts where the properties are unknown, controlled externally, and not validated.

Reply:
changed to an object after previous review

Reply:
Empty/Unschematized object is not allowed. This object needs to be defined.

Reply:
done. please see most recent revision.